src/v/cluster/bootstrap_backend.cc

```diff
@@ -203,7 +203,14 @@ bootstrap_backend::apply(bootstrap_cluster_cmd cmd, model::offset offset) {
           [o = offset,
            m = cmd.value.recovery_state->manifest,
            b = cmd.value.recovery_state->bucket](auto& recovery_table) {
-              recovery_table.apply(o, m, b, wait_for_nodes::yes);
+              auto ec = recovery_table.apply(o, m, b, wait_for_nodes::yes);
```

Comment:
> This shouldn't happen

throwing sounds good to me, but it sounds like we could also vassert no failure?

Reply:
cc @ztlpn

Reply:
I'm okay with that I had talked to @andrwng about this

Reply:
Right, we don't expect to get here since this is happening at bootstrap, and recoveries aren’t allowed to be started if we haven’t bootstrapped yet.

That said, if we want to be more conservative, we could alternatively handle `errc::update_in_progress` specifically like we do earlier in the function when we bootstrap users.

Reply:
I think I'll leave this - if someone else feels strongly feel free to send a followup patch.

Reply:
Right, throwing from an apply method is pretty nasty - if this happens when a snapshot is applied, the exception will be converted to an assert, and if this is a regular command, controller stm fiber will stall, leaving the node in a weird state (but at least operational). Pick your poison.

OTOH this method throws in other places, so I guess the patch is not making matters worse :)

Reply:
@andrwng sent a followup to change this: https://github.com/redpanda-data/redpanda/pull/17563